src/libbson/CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 3.1)
+cmake_minimum_required (VERSION 3.15)
```

Comment:
This is a significant change. I suggest at least creating a CXX ticket to track/document this new requirement.

Side question: is there a compelling reason we should continue to support a per-library CMake Project configuration instead of using only the root CMake project for all targets defined in the CXX Driver? I don't think we intend to support `add_subdirectory()` of a CXX Driver library (i.e. bsoncxx) _without_ going through the root directory, do we?

Reply:
We're already requiring this at the top level, so this line could likely actually just be removed. It's actually probably bad that we had `3.1` there, since it would rewind a bunch of CMake policies and probably had some unintentional effects. (Such as the problem in ResSearch).

Reply:
> I don't think we intend to support `add_subdirectory()` of a CXX Driver library (i.e. bsoncxx) _without_ going through the root directory, do we?

AFAIK there is no expectation to support `add_subdirectory()` without going through the root directory.

I expect C++ driver can be included with `add_subdirectory` of the root CMakeLists.txt, and CXX-2104 was filed to document.

This PR is for the C driver (not C++).

Reply:
Had my repos mixed up. Sorry for the confusion.